racker/salus-event-engine-management
# add labelSelector matching to kapacitor

# Resolves

SALUS-193 https://jira.rax.io/browse/SALUS-193

# What

It adds labels to all of the relevant objects and then decomposes those labels into a comparison that kapacitor can accomplish

# How

It configures the TICKScript template to do the comparisons inside of kapacitor. 

## How to test

Right now you should import @itzg 's new insomnia profile. 
Start event-engine-management, event-engine-ingest, and bring up the docker components.
Then create a new Task Object from geoffs insomnia profile with
`"labelSelector": {"os": "linux"}` tacked onto the end of the "taskParameters" section of the JSON body.

# Why
Kapacitor has very limited capacity to do comparisons. We need to do the comparison in kapacitor so that if the labels get updated somewhere else then we automatically get the changes through the data pipeline. 

# TODO

The TICK script is not seeing the values that I am trying to tell it to reference. So that needs to be fixed
